test/master-playlist-controller.test.js

```diff
@@ -2512,6 +2531,23 @@ QUnit.test('calculates dynamic GOAL_BUFFER_LENGTH', function(assert) {
 
   assert.equal(mpc.goalBufferLength(), 60, 'dynamic GBL continues to use max value');
 
+  mpc.setCurrentTime(120);
+  currentTime = 120;
+
+  assert.equal(mpc.goalBufferLength(), 30, 'dynamic GBL uses starting value at 0s after seek');
+
+  currentTime = 130;
+
+  assert.equal(mpc.goalBufferLength(), 35, 'dynamic GBL increases by elapsedSinceStart() * rate');
+
+  currentTime = 180;
+
+  assert.equal(mpc.goalBufferLength(), 60, 'dynamic GBL uses max value at 60s after seek');
```

Comment:
This is technically not after a seek, right?

Reply:
Care to elaborate? This test exists to make sure the dynamic GBL works as expected after a seek. I'm making the assumption in this PR that a seek will clear the buffer, so we should treat that as if we've started fresh. So here I'm seeking via `setCurrentTime` and testing to see that GBL responds in the same way as it does when starting from a fresh playback.

Reply:
My mistake. I saw the seek a couple tests higher and assumed the test after had to do with the seek, but then following tests had to do with just normal playback from that point (since only currentTime changed, but not via `setCurrentTime` again), but you're right, it's still after a seek.